nix/haskell.nix

```diff
@@ -0,0 +1,76 @@
+############################################################################
+# Builds Haskell packages with Haskell.nix
+############################################################################
+{ lib
+, stdenv
+, haskell-nix
+, buildPackages
+, config ? {}
+# GHC attribute name
+, compiler ? config.haskellNix.compiler or "ghc865"
+# Enable profiling
+, profiling ? config.haskellNix.profiling or false
+}:
+let
+
+  # This creates the Haskell package set.
+  # https://input-output-hk.github.io/haskell.nix/user-guide/projects/
+  pkgSet = haskell-nix.cabalProject {
+    src = haskell-nix.haskellLib.cleanGit {
+      name = "decentralized-software-updates";
+      src = ../.;
+    };
+    ghc = buildPackages.haskell-nix.compiler.${compiler};
+    modules = [
+
+      # Allow reinstallation of Win32
+      { nonReinstallablePkgs =
+        [ "rts" "ghc-heap" "ghc-prim" "integer-gmp" "integer-simple" "base"
+          "deepseq" "array" "ghc-boot-th" "pretty" "template-haskell"
+          # ghcjs custom packages
+          "ghcjs-prim" "ghcjs-th"
+          "ghc-boot"
+          "ghc" "array" "binary" "bytestring" "containers"
+          "filepath" "ghc-boot" "ghc-compact" "ghc-prim"
+          # "ghci" "haskeline"
+          "hpc"
+          "mtl" "parsec" "text" "transformers"
+          "xhtml"
+          # "stm" "terminfo"
+        ];
+      }
+      {
+        # Packages we wish to ignore version bounds of.
+        # This is similar to jailbreakCabal, however it
+        # does not require any messing with cabal files.
+        packages.katip.doExactConfig = true;
+
+        # split data output for ekg to reduce closure size
+        packages.ekg.components.library.enableSeparateDataOutput = true;
+        packages.cardano-node.configureFlags = [ "--ghc-option=-Werror" ];
+        packages.cardano-config.configureFlags = [ "--ghc-option=-Werror" ];
+        enableLibraryProfiling = profiling;
+      }
+      (lib.optionalAttrs stdenv.hostPlatform.isWindows {
+        # Disable cabal-doctest tests by turning off custom setups
+        packages.comonad.package.buildType = lib.mkForce "Simple";
```

Comment:
Are all these overrides required by the project? and if not, do we typically include them all anyway?